BackendServer/python/analysis.py

```diff
@@ -0,0 +1,163 @@
+
+    # Calculate grid points
+    output_df['Auto_Grid_Points'] = 3 * (output_df['Auto_Cube_Low'] + output_df['Auto_Cone_Low']) + 4 * (output_df['Auto_Cube_Mid'] + output_df['Auto_Cone_Mid']) + 6 * (output_df['Auto_Cube_High'] + output_df['Auto_Cone_High'])
+    output_df['Tele_Grid_Points'] =  2 * output_df['Tele_Low_Cycles'] + 3 * output_df['Tele_Mid_Cycles'] + 5 * output_df['Tele_High_Cycles']
+    
+    # Link points
+    # output_df['Assumed_Link_Points'] = 5 / 3 * (output_df['Auto_Cube_Low'] + output_df['Auto_Cone_Low'] + output_df['Tele_Cube_Low'] + output_df['Tele_Cone_Low'] + output_df['Auto_Cube_Mid'] + output_df['Auto_Cone_Mid'] + output_df['Tele_Cube_Mid'] + output_df['Tele_Cone_Mid'] + output_df['Auto_Cube_High'] + output_df['Auto_Cone_High'] + output_df['Tele_Cube_High'] + output_df['Tele_Cone_High'])
+    output_df['Assumed_Link_Points'] = 0
+
+    # Calculate mobility/station points
+    output_df['Mobility_Points'] = output_df['Mobility'].map({0: 0, 1: 3})
+    output_df['Auto_Station_Points'] = output_df['Auto_Station'].map({AutoStation.noPoints: 0, AutoStation.failed: 0, AutoStation.docked: 8, AutoStation.engaged: 12})
+    output_df['Tele_Station_Points'] = output_df['Tele_Station'].map({TeleStation.noPoints: 0, TeleStation.parked: 0, TeleStation.failed: 0, TeleStation.docked: 8, TeleStation.engaged: 12})
+
+    # Total points
+    output_df['Total_Auto_Points'] = output_df['Auto_Grid_Points'] + output_df['Auto_Station_Points'] + output_df['Mobility_Points']
+    output_df['Total_Tele_Points'] = output_df['Tele_Grid_Points'] + output_df['Tele_Station_Points'] + output_df['Assumed_Link_Points']
+    output_df['Total_Points'] = output_df['Total_Auto_Points'] + output_df['Total_Tele_Points']
+    
+    return output_df
+
+def calculate_match_analysis(Team_Number, db_connection, *, appendTo = {}):
+    # Read data from MySQL database
+    matches_df = pd.read_sql('SELECT * FROM matchData WHERE Team_Number = %s AND No_Show_Robot = FALSE', db_connection, params=(Team_Number,))
+
+    # Calculate additional scores
+    calculate_match_scores(matches_df, mutate=True)
+    
+    # Create output variable
+    analysis_output = appendTo
+    
+    # Insert team number
+    analysis_output['Team_Number'] = Team_Number
+
+    # Calculate scoring aggregates
+    for type_phase, prefixes in (
+        ('Auto',        ('Auto_Cone', 'Auto_Cube')),
+        ('Tele_Pieces', ('Tele_Cone', 'Tele_Cube')),
+        ('Tele_Cone',   ('Tele_Cone',)),
+        ('Tele_Cube',   ('Tele_Cube',))
+    ):
+        for level, suffixes in (
+            ('Total', ('Low', 'Mid', 'High')),
+            ('Low',   ('Low',)),
+            ('Mid',   ('Mid',)),
+            ('High',  ('High',))
+        ):
+            # Get names of all the columns we want
+            included_columns = [f"{prefix}_{suffix}" for prefix in prefixes for suffix in suffixes]
+
+            # Sum up all of the included columns, then find the mean or max of it
```

Comment:
Even though we aren't using it now, what are your thoughts on performing the `min` as well here? We probably won't use it for any statistical analysis, but it could be used for graphics/plots in the future

Reply:
Sure, I'll just add more database columns